obspy/io/gse2/bulletin.py

```diff
@@ -0,0 +1,1117 @@
+                    if begin_block:
+                        # 2 BEGIN without STOP
+                        message = self._add_line_nb('Missing STOP tag')
+                        raise GSE2BulletinSyntaxError(message)
+                    else:
+                        # Enter a BEGIN block
+                        begin_block = True
+
+                    self._check_header(line)
+                elif line.startswith('STOP'):
+                    if begin_block:
+                        # Exit a BEGIN-STOP block
+                        begin_block = False
+                    else:
+                        # STOP without BEGIN
+                        message = self._add_line_nb('Missing BEGIN tag')
+                        raise GSE2BulletinSyntaxError(message)
+                elif line.startswith('DATA_TYPE'):
+                    bulletin_block = line[10:18] == 'BULLETIN'
+
+                if not begin_block or not bulletin_block:
+                    # Not in a BEGIN-STOP block, nor a DATA_TYPE BULLETIN
+                    # block.
+                    continue
+
+                # If a "Reviewed Event Bulletin" or "Reviewed Bulletin"
+                # line exists, put it in comment
+                if 'Reviewed Event Bulletin' in line \
+                        or 'Reviewed Bulletin' in line:
+                    comment = self._comment(line.strip())
+                    if comment.text:
+                        catalog.comments.append(comment)
+                # Detect start of an event
+                elif line.startswith('EVENT'):
+                    event = self._parse_event(line)
+                    if event:
+                        catalog.append(event)
+
+        except StopIteration:
+            message = self._add_line_nb('Unexpected EOF while parsing')
+            raise GSE2BulletinSyntaxError(message)
+        except:
+            self._warn('Unexpected error')
+            raise
+
+        if begin_block:
+            # BEGIN-STOP block not closed
+            text = 'Unexpected EOF while parsing, BEGIN-STOP block not closed'
+            message = self._add_line_nb(text)
+            raise GSE2BulletinSyntaxError(message)
+
+        catalog.resource_id = self._get_res_id('event/evid')
+
+        return catalog
+
+
+def _read_gse2(filename, inventory=None, default_network_code='XX',
+               default_location_code=None, default_channel_code=None,
+               res_id_prefix='smi:local', fields=None,
+               event_point_separator=False, agency=None):
+    """
+    Read a GSE2.0 bulletin file and returns an ObsPy Catalog object.
+
+    .. warning::
+        This function should NOT be called directly, it registers via the
+        ObsPy :func:`~obspy.core.event.read_events` function, call this
+        instead.
+
+    :type filename: str
+    :param filename: File or file-like object in text mode.
+    :type inventory: :class:`~obspy.core.inventory.inventory.Inventory`,
+        optionnal
```

Comment:
Typo. But please just remove the "optional" from all parameters. I know we still have it some old functions/methods but it is superfluous as the function signature provides the same information.